### PR TITLE
Refactor Skyview module to use pyVO's SIAP

### PR DESCRIPTION
**Pull request by duytnguyendtn**
Hello Astroquery Maintainers!

This PR offers to refactor the Skyview module, currently webscraping parameters using our Perl script, to instead use the Virtual Observatory's Simple Image Access Protocol via the pyVO package. Feedback and suggestions would be highly welcome!

There were some notable changes that I made that is different from the original Skyview class, including:
1. The reported parameters the Skyview surveys support are much smaller in scope than the full capabilities of the original Astroquery Skyview package. Removed parameters include: the `resolver`, `deedger`, `lut`, `grid`, and `gridlabels` arguments. (Commit bd6864c4b4faeb13357607fa82d4f88539043051)
2. The online Perl form only supports specifying the size of the FOV in degrees, rather than through width and height. Because the Astroquery spec states only one is required, I brought the skyview API in line with the Perl script and only provided `radius` as a parameter, dropping the width/height args. (Commit 9e83dd0a69f8b4e01ac22c3c1f64872943ca753c)
3. ~To make it more explicit, I renamed the `survey` arg to `surveys` (with an s) to indicate the user may specify multiple surveys.~ Reverted (see https://github.com/astropy/astroquery/pull/2849#discussion_r1350815065)

Tests have been updated as well, though I did run into an issue with one of the remote tests, where the download is timing out, but there is no way for us to override the timeout from the level of the astroquery module. (If I run the remote tests consecutively (4-5 times), the test passes presumably after some of the files are cached somewhere between my computer and the server). I reported it in #astroquery on slack (copied here for reference):

<details>

Hi all, undergoing a similar effort as Brigitta and refactoring the SkyView class to use pyVO's SIA. I'm hitting a timeout error trying to download the FITS files from one of our larger archives to open them with `astropy.io.fits`. I'm using `astroquery.utils.commons.FileContainer.get_fits` , but there doesn't seem to be a way to control the timeout. Is the recommended way to handle this the same way `astroquery.besancon` does and call `astroquery.utils.commons.get_readable_fileobj` directly to [handle the timeout manually](https://github.com/astropy/astroquery/blob/8fe1a8b283905796e8a3da1f7765fcd7912d0b92/astroquery/besancon/core.py#L114-L144), or is there a better way?

</details>

This PR is still in draft until I complete the documentation changes. Thank you!

### CODE REVIEW

**Review by bsipocz — COMMENTED** [review]
@duytnguyendtn  - Thanks for the PR.

There are a lot of code back and forth in the commit history, I don't think it has any benefits to keep those duplicated classes around. Could you please squash them out?

Besides that, as the first round I left some minimal comments, will come back at a deeper look when you flip it out from the draft status. 

For the kwarg deprecations, have a look into the IRSA module and its usage of `deprecated_renamed_argument`.

File: astroquery/skyview/tests/test_skyview_remote.py [hunk 1]
```diff
@@ -1,52 +1,24 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import pytest
-import json
 
 from astropy.io.fits import HDUList
 
 from ...skyview import SkyView
-from .test_skyview import data_path
+from . import ETA_CARINAE_MULTI_CASE_ARGS, ETA_CARINAE_SINGLE_CASE_ARGS
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astroquery/skyview/tests/test_skyview_remote.py]: Done! I originally put this here thinking I would also use the same usecase for my local tests, but once I mocked up the entire response, this wasn't necessary

File: astroquery/skyview/core.py [hunk 2]
```diff
@@ -148,38 +80,8 @@ def get_images(self, position, survey, *, coordinates=None, projection=None,
         sampler : str
             The sampling algorithm determines how the data requested will
             be resampled so that it can be displayed.
-        resolver : str
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astroquery/skyview/core.py:151]: Thanks for the reference to the IRSA code. I think I addressed this properly!

File: astroquery/skyview/core.py [hunk 3]
```diff
@@ -107,7 +39,7 @@ def get_images(self, position, survey, *, coordinates=None, projection=None,
             supported. Object names are converted to coordinates via the
             SIMBAD or NED name resolver. See the reference for more info
             on the supported syntax for coordinates.
-        survey : str or list of str
+        surveys : str or list of str
```
[the remark this answers is not in the document]
Comment (duytnguyendtn) [reply astroquery/skyview/core.py]: Reverted!

**Comment by duytnguyendtn** [comment]
@bsipocz Thanks for the comments! I think I addressed them and my documentation changes, so this PR should be ready for a full review. Thank you in advance!

Also pinging @zoghbi-a, another HEASARC member who may be interested in taking a look

**Comment by duytnguyendtn** [comment]
@bsipocz Any sense of when we might be able to get a review on this PR? Thanks!

**Review by keflavich — COMMENTED** [review]
This looks like a significant improvement in that it simplifies the interface and uses a supported API.  Does it fully replicate the old functionality?  At least some keywords, like `width` and `height`, are no longer supported.  The docs should at least give some clue about the difference between the web interface and the new API.

**Comment by duytnguyendtn** [comment]
Thanks for the comments @keflavich! Sadly no, this does not fully replace the old API due to the SIAP parameters not being a 1-1 match to the ones available via the online perl form. As per https://github.com/astropy/astroquery/pull/2849#discussion_r1350816301, I tried to properly deprecate them on the code side and explicitly list the supported parameters in the documentation in https://github.com/astropy/astroquery/pull/2849/commits/2a6135a3d410ff6c023b793f4b1e4444a69387fa.

Would it be appropriate to explicitly list the parameters being removed in a warning? Or are you instead thinking of a smaller guiding the user's attention to the difference in available parameters?

**Comment by keflavich** [comment]
The deprecation strategy will already do that, but I'm concerned: if we're losing functionality, then we shouldn't just straight up replace the old code; instead, these should live side-by-side.

However, it's not entirely clear to me that we do lose functionality. Yes, we lose keywords, but do those keywords have equivalents in SIAP?  i.e., can the same images be retrieved with a different set of keywords?  If so, we should provide users with at least some examples of how that would look.

File: astroquery/skyview/core.py [hunk 4]
```diff
@@ -241,95 +156,89 @@ def get_image_list(self, position, survey, *, coordinates=None,
...
-
-    def _parse_response(self, response):
-        bs = BeautifulSoup(response.content, "html.parser")
         urls = []
-        for a in bs.find_all('a'):
-            if a.text == 'FITS':
-                href = a.get('href')
-                urls.append(urlparse.urljoin(response.url, href))
+        for current_survey in survey:
+            registry_results = registry.search(ivoid=self._survey_dict[current_survey]['ivoid'])
```
Comment (zoghbi-a) [inline astroquery/skyview/core.py:169]: I don't think there is a need for another call to the registry here. During the first call in `_survey_dict`, you can also save `access_url`, and then use it here to create the ` sia_service` by something like: ` sia_service  = pyvo.sia.SIAService(self._survey_dict[current_survey]['access_url'])`

**Review by bsipocz — CHANGES_REQUESTED** [review]
The big review point here is to remove all, or at least some of the registry calls. It adds complexity and we don't really need to do multi-archive data discovery here, but only plan to do the skyview services.

For both other cases where we do a `query_sia`, we do that directly, without doing any registry search. Could we follow it here, too rather than the more complex workflow of calling the registry and then do the search?

I would ping @andamian here to see whether he agrees as the alma module has the functionality the longest.

**Comment by duytnguyendtn** [comment]
Hi all, thanks for your patience! Coming back to this after a bit of hiatus, I spoke with members of the HEASARC regarding how to move forward and we think we'd like to refactor our SIA service on our end first to support the missing arguments described here. Because a lot of the deprecation work that was added to this branch won't be needed (and we don't have a timeline to when this SIA work will be complete), I'm going to close this PR to get this off of your radar and reintroduce it once we (A) finish supporting the missing arguments and (B) I address the above comments the Astroquery team has made.

Thanks again! I'll be back with a second attempt at this!

**Pull request closed without merging**